<ui version="4.0" >
 <class>Editor</class>
 <widget class="QWidget" name="Editor" >
  <property name="geometry" >
   <rect>
    <x>0</x>
    <y>0</y>
    <width>447</width>
    <height>392</height>
   </rect>
  </property>
  <layout class="QGridLayout" >
   <property name="margin" >
    <number>0</number>
   </property>
   <item row="0" column="0" >
    <widget class="QTabWidget" name="tabWidget" >
     <property name="currentIndex" >
      <number>0</number>
     </property>
     <widget class="QWidget" name="tab" >
      <attribute name="title" >
       <string>&amp;General</string>
      </attribute>
      <layout class="QVBoxLayout" >
       <item>
        <layout class="QGridLayout" >
         <item row="0" column="0" >
          <widget class="QLabel" name="label" >
           <property name="minimumSize" >
            <size>
             <width>80</width>
             <height>0</height>
            </size>
           </property>
           <property name="text" >
            <string>Site Name</string>
           </property>
           <property name="alignment" >
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
         <item row="0" column="1" >
          <widget class="KLineEdit" name="siteName" >
           <property name="font" >
            <font>
             <weight>75</weight>
             <bold>true</bold>
            </font>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QGridLayout" >
         <item row="0" column="0" >
          <widget class="QLabel" name="label_6" >
           <property name="minimumSize" >
            <size>
             <width>80</width>
             <height>0</height>
            </size>
           </property>
           <property name="text" >
            <string>Protocol</string>
           </property>
           <property name="alignment" >
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
         <item row="0" column="1" >
          <widget class="KComboBox" name="protocol" >
           <item>
            <property name="text" >
             <string>FTP</string>
            </property>
           </item>
           <item>
            <property name="text" >
             <string>SFTP (via SSH)</string>
            </property>
           </item>
          </widget>
         </item>
         <item row="1" column="0" >
          <widget class="QLabel" name="label_2" >
           <property name="minimumSize" >
            <size>
             <width>69</width>
             <height>0</height>
            </size>
           </property>
           <property name="text" >
            <string>IP Address</string>
           </property>
           <property name="alignment" >
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
         <item row="1" column="1" >
          <widget class="KLineEdit" name="ipAddress" />
         </item>
         <item row="1" column="2" >
          <layout class="QHBoxLayout" >
           <item>
            <widget class="QLabel" name="label_4" >
             <property name="sizePolicy" >
              <sizepolicy vsizetype="Preferred" hsizetype="Fixed" >
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text" >
              <string>Port</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="port" >
             <property name="buttonSymbols" >
              <enum>QAbstractSpinBox::NoButtons</enum>
             </property>
             <property name="minimum" >
              <number>1</number>
             </property>
             <property name="maximum" >
              <number>65565</number>
             </property>
             <property name="value" >
              <number>21</number>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item row="2" column="0" >
          <widget class="QLabel" name="label_3" >
           <property name="text" >
            <string>User Name</string>
           </property>
           <property name="alignment" >
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
         <item row="2" column="1" >
          <widget class="KLineEdit" name="userName" />
         </item>
         <item row="2" column="2" >
          <widget class="QCheckBox" name="anonymous" >
           <property name="text" >
            <string>Anonymous</string>
           </property>
          </widget>
         </item>
         <item row="3" column="0" >
          <widget class="QLabel" name="label_5" >
           <property name="text" >
            <string>Password</string>
           </property>
           <property name="alignment" >
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
         <item row="3" column="1" >
          <widget class="KLineEdit" name="password" >
           <property name="passwordMode" >
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QGridLayout" >
         <item row="0" column="0" >
          <widget class="QLabel" name="label_7" >
           <property name="minimumSize" >
            <size>
             <width>80</width>
             <height>0</height>
            </size>
           </property>
           <property name="text" >
            <string>Remote Path</string>
           </property>
           <property name="alignment" >
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
         <item row="0" column="1" >
          <widget class="KLineEdit" name="remotePath" />
         </item>
         <item row="1" column="0" >
          <widget class="QLabel" name="label_8" >
           <property name="text" >
            <string>Local Path</string>
           </property>
           <property name="alignment" >
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
         <item row="1" column="1" >
          <widget class="KUrlRequester" name="localPath" />
         </item>
         <item row="2" column="0" >
          <layout class="QVBoxLayout" >
           <item>
            <widget class="QLabel" name="label_9" >
             <property name="text" >
              <string>Notes</string>
             </property>
             <property name="alignment" >
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
           <item>
            <spacer>
             <property name="orientation" >
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0" >
              <size>
               <width>20</width>
               <height>40</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
         <item row="2" column="1" >
          <widget class="QTextEdit" name="notes" />
         </item>
        </layout>
       </item>
       <item>
        <spacer>
         <property name="orientation" >
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0" >
          <size>
           <width>20</width>
           <height>0</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_2" >
      <attribute name="title" >
       <string>Options</string>
      </attribute>
      <layout class="QVBoxLayout" >
       <property name="spacing" >
        <number>5</number>
       </property>
       <item>
        <widget class="QCheckBox" name="disableExtendedPassive" >
         <property name="text" >
          <string>Disable use of extended passive mode</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="disablePassive" >
         <property name="text" >
          <string>Disable use of passive mode</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="useSiteIp" >
         <property name="text" >
          <string>Use site IP for passive mode connections</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="disablePresetIp" >
         <property name="text" >
          <string>Disable pre-set active mode IP for this site</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="useStat" >
         <property name="text" >
          <string>Use faster directory listing via STAT</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QCheckBox" name="disableThreads" >
         <property name="text" >
          <string>Do not use multiple transfer threads for this site</string>
         </property>
        </widget>
       </item>
       <item>
        <spacer>
         <property name="orientation" >
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0" >
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_3" >
      <attribute name="title" >
       <string>Advanced</string>
      </attribute>
      <layout class="QVBoxLayout" >
       <item>
        <layout class="QGridLayout" >
         <item row="0" column="0" >
          <widget class="QLabel" name="label_10" >
           <property name="text" >
            <string>Server Encoding</string>
           </property>
           <property name="alignment" >
            <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
           </property>
          </widget>
         </item>
         <item row="0" column="1" >
          <widget class="KComboBox" name="encoding" >
           <property name="sizePolicy" >
            <sizepolicy vsizetype="Fixed" hsizetype="Expanding" >
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
          </widget>
         </item>
         <item row="1" column="0" >
          <layout class="QVBoxLayout" >
           <item>
            <widget class="QLabel" name="label_11" >
             <property name="text" >
              <string>Retries</string>
             </property>
             <property name="alignment" >
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
           <item>
            <spacer>
             <property name="orientation" >
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0" >
              <size>
               <width>20</width>
               <height>0</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
         <item row="1" column="1" >
          <widget class="QGroupBox" name="retry" >
           <property name="sizePolicy" >
            <sizepolicy vsizetype="Fixed" hsizetype="Preferred" >
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="title" >
            <string>Retry to connect on failure</string>
           </property>
           <property name="flat" >
            <bool>true</bool>
           </property>
           <property name="checkable" >
            <bool>true</bool>
           </property>
           <property name="checked" >
            <bool>false</bool>
           </property>
           <layout class="QGridLayout" >
            <item row="0" column="0" colspan="2" >
             <widget class="QLabel" name="label_12" >
              <property name="text" >
               <string>Retry delay</string>
              </property>
             </widget>
            </item>
            <item row="0" column="2" >
             <widget class="QSpinBox" name="retryDelay" >
              <property name="sizePolicy" >
               <sizepolicy vsizetype="Fixed" hsizetype="Fixed" >
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimumSize" >
               <size>
                <width>55</width>
                <height>0</height>
               </size>
              </property>
              <property name="minimum" >
               <number>10</number>
              </property>
              <property name="maximum" >
               <number>9999</number>
              </property>
              <property name="value" >
               <number>60</number>
              </property>
             </widget>
            </item>
            <item rowspan="2" row="1" column="0" colspan="2" >
             <widget class="QLabel" name="label_13" >
              <property name="text" >
               <string>Number of retries (0 = infinite)</string>
              </property>
             </widget>
            </item>
            <item rowspan="2" row="1" column="2" >
             <widget class="QSpinBox" name="retryCount" >
              <property name="sizePolicy" >
               <sizepolicy vsizetype="Fixed" hsizetype="Fixed" >
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimumSize" >
               <size>
                <width>55</width>
                <height>0</height>
               </size>
              </property>
              <property name="value" >
               <number>10</number>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item row="2" column="0" >
          <layout class="QVBoxLayout" >
           <item>
            <widget class="QLabel" name="label_14" >
             <property name="text" >
              <string>Keepalive</string>
             </property>
             <property name="alignment" >
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
           <item>
            <spacer>
             <property name="orientation" >
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0" >
              <size>
               <width>20</width>
               <height>40</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
         <item row="2" column="1" >
          <widget class="QGroupBox" name="keepalive" >
           <property name="sizePolicy" >
            <sizepolicy vsizetype="Fixed" hsizetype="Preferred" >
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="title" >
            <string>Use keepalive to keep the connection open</string>
           </property>
           <property name="flat" >
            <bool>true</bool>
           </property>
           <property name="checkable" >
            <bool>true</bool>
           </property>
           <property name="checked" >
            <bool>false</bool>
           </property>
           <layout class="QGridLayout" >
            <item row="0" column="0" >
             <widget class="QLabel" name="label_15" >
              <property name="text" >
               <string>Frequency (seconds)</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1" >
             <widget class="QSpinBox" name="keepaliveFrequency" >
              <property name="sizePolicy" >
               <sizepolicy vsizetype="Fixed" hsizetype="Fixed" >
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="minimumSize" >
               <size>
                <width>55</width>
                <height>0</height>
               </size>
              </property>
              <property name="minimum" >
               <number>30</number>
              </property>
              <property name="maximum" >
               <number>600</number>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <spacer>
         <property name="orientation" >
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0" >
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_4" >
      <attribute name="title" >
       <string>Security</string>
      </attribute>
      <layout class="QGridLayout" >
       <item row="0" column="0" >
        <widget class="QStackedWidget" name="securityStack" >
         <property name="currentIndex" >
          <number>0</number>
         </property>
         <widget class="QWidget" name="page" >
          <layout class="QGridLayout" >
           <item row="0" column="1" >
            <spacer>
             <property name="orientation" >
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0" >
              <size>
               <width>20</width>
               <height>40</height>
              </size>
             </property>
            </spacer>
           </item>
           <item row="1" column="0" >
            <spacer>
             <property name="orientation" >
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0" >
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item row="1" column="1" >
            <widget class="QLabel" name="label_16" >
             <property name="minimumSize" >
              <size>
               <width>312</width>
               <height>0</height>
              </size>
             </property>
             <property name="text" >
              <string>&lt;i>The protocol currently selected for this site does not support any security configuration options.&lt;/i></string>
             </property>
             <property name="wordWrap" >
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item row="1" column="2" >
            <spacer>
             <property name="orientation" >
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0" >
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item row="2" column="1" >
            <spacer>
             <property name="orientation" >
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0" >
              <size>
               <width>20</width>
               <height>40</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="page_2" >
          <layout class="QGridLayout" >
           <item row="0" column="0" >
            <widget class="QLabel" name="label_20" >
             <property name="text" >
              <string>SSL Negotiation Type</string>
             </property>
             <property name="alignment" >
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
           <item row="0" column="1" >
            <widget class="QComboBox" name="sslNegotiationType" >
             <item>
              <property name="text" >
               <string>None (plain FTP)</string>
              </property>
             </item>
             <item>
              <property name="text" >
               <string>AUTH TLS</string>
              </property>
             </item>
             <item>
              <property name="text" >
               <string>Implicit SSL</string>
              </property>
             </item>
            </widget>
           </item>
           <item row="1" column="0" >
            <widget class="QLabel" name="label_21" >
             <property name="text" >
              <string>Data Protection</string>
             </property>
             <property name="alignment" >
              <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
             </property>
            </widget>
           </item>
           <item row="1" column="1" >
            <widget class="QComboBox" name="sslProtMode" >
             <item>
              <property name="text" >
               <string>Always encrypt the data channel</string>
              </property>
             </item>
             <item>
              <property name="text" >
               <string>Encrypt only directory listings</string>
              </property>
             </item>
             <item>
              <property name="text" >
               <string>Do not encrypt the data channel</string>
              </property>
             </item>
            </widget>
           </item>
           <item row="2" column="0" colspan="2" >
            <widget class="QGroupBox" name="sslClientCert" >
             <property name="title" >
              <string>Use client SSL certificate for authentication</string>
             </property>
             <property name="flat" >
              <bool>true</bool>
             </property>
             <property name="checkable" >
              <bool>true</bool>
             </property>
             <property name="checked" >
              <bool>false</bool>
             </property>
             <layout class="QGridLayout" >
              <item row="0" column="0" >
               <widget class="QLabel" name="label_17" >
                <property name="text" >
                 <string>Certificate</string>
                </property>
               </widget>
              </item>
              <item row="0" column="1" >
               <widget class="KUrlRequester" name="sslCertPath" />
              </item>
              <item row="1" column="0" >
               <widget class="QLabel" name="label_19" >
                <property name="text" >
                 <string>Key file</string>
                </property>
               </widget>
              </item>
              <item row="1" column="1" >
               <widget class="KUrlRequester" name="sslKeyPath" />
              </item>
             </layout>
            </widget>
           </item>
           <item row="3" column="0" colspan="2" >
            <spacer>
             <property name="orientation" >
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0" >
              <size>
               <width>20</width>
               <height>40</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="page_3" >
          <layout class="QVBoxLayout" >
           <item>
            <widget class="QGroupBox" name="sshPubkey" >
             <property name="title" >
              <string>Use public-key authentication</string>
             </property>
             <property name="flat" >
              <bool>true</bool>
             </property>
             <property name="checkable" >
              <bool>true</bool>
             </property>
             <property name="checked" >
              <bool>false</bool>
             </property>
             <layout class="QGridLayout" >
              <item row="0" column="0" >
               <widget class="QLabel" name="label_18" >
                <property name="text" >
                 <string>Public Key</string>
                </property>
               </widget>
              </item>
              <item row="0" column="1" >
               <widget class="KUrlRequester" name="sshPublicKey" />
              </item>
              <item row="1" column="0" >
               <widget class="QLabel" name="label_22" >
                <property name="text" >
                 <string>Private Key</string>
                </property>
               </widget>
              </item>
              <item row="1" column="1" >
               <widget class="KUrlRequester" name="sshPrivateKey" />
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <spacer>
             <property name="orientation" >
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0" >
              <size>
               <width>407</width>
               <height>181</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </widget>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>KComboBox</class>
   <extends>QComboBox</extends>
   <header>kcombobox.h</header>
  </customwidget>
  <customwidget>
   <class>KLineEdit</class>
   <extends>QLineEdit</extends>
   <header>klineedit.h</header>
  </customwidget>
  <customwidget>
   <class>KUrlRequester</class>
   <extends>QFrame</extends>
   <header>kurlrequester.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>anonymous</sender>
   <signal>toggled(bool)</signal>
   <receiver>password</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>392</x>
     <y>139</y>
    </hint>
    <hint type="destinationlabel" >
     <x>316</x>
     <y>165</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>anonymous</sender>
   <signal>toggled(bool)</signal>
   <receiver>userName</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>365</x>
     <y>137</y>
    </hint>
    <hint type="destinationlabel" >
     <x>311</x>
     <y>139</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
